Remove coreboot detection, add skip_grub parameter

rustybird opened the pull request:
Successfully tested the following scenarios with the R3.2 iso:

- `skip_grub=1`, following instructions from https://github.com/QubesOS/qubes-issues/issues/2118#issue-162353768
- No `skip_grub` parameter, default installation layout
- No `skip_grub` parameter, btrfs installation layout (generated by Anaconda)

The dmidecode package could be removed now, but they still seem like useful tools to have for debugging installer problems.

marmarek commented on anaconda/pyanaconda/bootloader.py:
```diff
@@ -1419,13 +1419,9 @@ class GRUB2(GRUB):
-            self.stage2_format_types += ["lvmlv"]
+        self.encryption_support = True
+        self.stage2_format_types += ["lvmlv"]
+        self.skip_bootloader = bool(int(flags.cmdline.get("skip_grub") or "0"))
```
Better use `flags.cmdline.getbool("skip_grub", False)`

Commit:
Remove coreboot detection, add skip_grub parameter

And unconditionally allow boot encryption and the lvmlv format.
(The user still has to fight the installer to actually set it up.)

Fixes QubesOS/qubes-issues#2553

rustybird commented:
Fixed. Booting with/without `skip_grub` assigns the expected values to `self.skip_bootloader`.
